x/oracle/keeper/historic_price.go

```diff
@@ -0,0 +1,201 @@
+package keeper
+
+import (
+	"sort"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/umee-network/umee/v3/x/oracle/types"
+)
+
+// median returns the median of a list of historic prices.
+func median(prices []types.HistoricPrice) sdk.Dec {
+	lenPrices := len(prices)
+	if lenPrices == 0 {
+		return sdk.ZeroDec()
+	}
+
+	sort.Slice(prices, func(i, j int) bool {
+		return prices[i].ExchangeRates.ExchangeRate.BigInt().
+			Cmp(prices[j].ExchangeRates.ExchangeRate.BigInt()) > 0
+	})
```

Comment:
So we sort by rates, but what if the rates are the same? Is it randomized order?

Reply:
Median is median. It's OK to have same values in a sequence.